transmission_interface/src/bidirectional_effort_joint_interface_provider.cpp

```diff
@@ -0,0 +1,87 @@
+///////////////////////////////////////////////////////////////////////////////
+// Copyright (C) 2013, PAL Robotics S.L.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//   * Redistributions of source code must retain the above copyright notice,
+//     this list of conditions and the following disclaimer.
+//   * Redistributions in binary form must reproduce the above copyright
+//     notice, this list of conditions and the following disclaimer in the
+//     documentation and/or other materials provided with the distribution.
+//   * Neither the name of PAL Robotics S.L. nor the names of its
+//     contributors may be used to endorse or promote products derived from
+//     this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+//////////////////////////////////////////////////////////////////////////////
+
+// Pluginlib
+#include <pluginlib/class_list_macros.h>
+
+// ros_control
+#include <transmission_interface/bidirectional_effort_joint_interface_provider.h>
+
+namespace transmission_interface
+{
+    bool BiDirectionalEffortJointInterfaceProvider::registerTransmission(TransmissionLoaderData &loader_data,
+            TransmissionHandleData &handle_data)
+    {
+        // Setup joint state interface first (if not yet done)
```

Comment:
This is still repeated from https://github.com/jlack1987/ros_control/blob/dfcb72375a3378b99e64971a46ef5f612f7dbbca/transmission_interface/src/effort_joint_interface_provider.cpp#L124

Why don't you do this only?

``` cpp
if(!JointStateInterfaceProvider::registerTransmission(loader_data, handle_data))
{
  return false;
}
```

The code in this line is done inside that method, and even worse, there's code in that method you're repeating a few lines below here. I'd mark below the bits you should remove here, since they're already in the other/parent one.